src/adv.py

```diff
@@ -32,20 +40,63 @@
 room['narrow'].w_to = room['foyer']
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
+room['treasure'].d_to = room['stairway']
+room['stairway'].u_to = room['treasure']
+room['stairway'].n_to = room['secretroom']
+room['secretroom'].s_to = room['stairway']
+room['secretroom'].k_to = room['realtreasureroom']
+
+player = Player(room['outside'])
+
+def direction_successful(direction, current_room):
+    attribute = direction + "_to"
+
+    if hasattr(current_room, attribute):
+        return getattr(current_room, attribute)
```

Comment:
I like how you separated this function out.  However, you could consider making this move function as a method on the player object.